addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -231,6 +231,10 @@ const threadPatch = {
     /** @override */
     open(options) {
         if (this.channel) {
+            if (this.self_member_id && !this.self_member_id.is_pinned) {
+                this.self_member_id.is_pinned = true;
+                this.channel.pinRpc({ pinned: true });
+            }
```

Comment:
For the discuss app, shouldn't it be part of `setAsDiscussThread`?

Basically we want to actually pin instead of writing `isLocallyPinned`?

With auto-pin `isLocallyPinned` basically becomes a live chat only feature.

Reply:
Well not so far, we still need `isLocallyPinned` for when not being a member.
But the comment still stands about doing it at the same place.